shell checker is linting files vendor folders

We should be skipping these files

https://github.com/knative-sandbox/net-contour/actions/runs/5159807027/jobs/9295187930?pr=920

```
Raw Output:
vendor/k8s.io/code-generator/generate-internal-groups.sh:129:-      --input-dirs "$(codegen::join , "${ALL_FQ_APIS[@]}")" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:130:-      --versioned-clientset-package "${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}/${CLIENTSET_NAME_VERSIONED:-versioned}" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:131:-      --internal-clientset-package "${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}/${CLIENTSET_NAME_INTERNAL:-internalversion}" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:132:-      --listers-package "${OUTPUT_PKG}/listers" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:133:-      --output-package "${OUTPUT_PKG}/informers" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:134:-      "$@"
vendor/k8s.io/code-generator/generate-internal-groups.sh:133:+    --input-dirs "$(codegen::join , "${ALL_FQ_APIS[@]}")" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:134:+    --versioned-clientset-package "${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}/${CLIENTSET_NAME_VERSIONED:-versioned}" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:135:+    --internal-clientset-package "${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}/${CLIENTSET_NAME_INTERNAL:-internalversion}" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:136:+    --listers-package "${OUTPUT_PKG}/listers" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:137:+    --output-package "${OUTPUT_PKG}/informers" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:138:+    "$@"
Warning: [shfmt] reported by reviewdog 🐶


Raw Output:
vendor/k8s.io/code-generator/generate-internal-groups.sh:141:-      --input-dirs "$(codegen::join , "${EXT_FQ_APIS[@]}" "${OPENAPI_EXTRA_PACKAGES[@]+"${OPENAPI_EXTRA_PACKAGES[@]}"}")" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:142:-      --input-dirs "k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/runtime,k8s.io/apimachinery/pkg/version" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:143:-      --output-package "${OUTPUT_PKG}/openapi" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:144:-      -O zz_generated.openapi \
vendor/k8s.io/code-generator/generate-internal-groups.sh:145:-      "$@"
vendor/k8s.io/code-generator/generate-internal-groups.sh:145:+    --input-dirs "$(codegen::join , "${EXT_FQ_APIS[@]}" "${OPENAPI_EXTRA_PACKAGES[@]+"${OPENAPI_EXTRA_PACKAGES[@]}"}")" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:146:+    --input-dirs "k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/runtime,k8s.io/apimachinery/pkg/version" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:147:+    --output-package "${OUTPUT_PKG}/openapi" \
vendor/k8s.io/code-generator/generate-internal-groups.sh:148:+    -O zz_generated.openapi \
vendor/k8s.io/code-generator/generate-internal-groups.sh:149:+    "$@"
Dropped refs/stash@{0} (a2dbf209ab941af992ecf88d148c1eff3d632d44)
Error: Process completed with exit code 1.
```

--- a/composite/style/shell/action.yml
+++ b/composite/style/shell/action.yml
@@ -15,6 +15,7 @@ runs:
     - id: shellcheck_suggester
       uses: mathiasvr/command-output@v2.0.0
       with:
+        # Remove vendor/ so actions don't try linting files in it
         run: |
           echo ${{ inputs.changed_shell_files }}
           for sh_file in ${{ inputs.changed_shell_files }}; do

--- a/composite/style/shell/action.yml
+++ b/composite/style/shell/action.yml
@@ -20,6 +20,8 @@ runs:
           for sh_file in ${{ inputs.changed_shell_files }}; do
             shellcheck -f diff $sh_file | patch -p1
           done
+
+          rm -rf vendor/
         shell: bash
       continue-on-error: true
     - name: suggester / shellcheck
@@ -39,6 +41,7 @@ runs:
       uses: reviewdog/action-shellcheck@v1
       with:
         reporter: 'github-pr-review'
+        exclude: './vendor/*'
         shellcheck_flags: |
           --external-sources -e=SC1091
 

--- a/composite/style/shell/action.yml
+++ b/composite/style/shell/action.yml
@@ -34,7 +34,7 @@ runs:
     - uses: reviewdog/action-shfmt@v1
       with:
         level: 'warning'
-        reporter: 'github-pr-review'
+        reviewdog_flags: -reporter=github-pr-review
 
     - name: "ShellCheck"
       if: ${{ steps.shellcheck_suggester.outputs.stderr }}

